app/routes/goal_routes.py

```diff
@@ -0,0 +1,121 @@
+from re import T
+from flask import Blueprint, json, jsonify, make_response, request, abort
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from datetime import datetime
+
+goal_bp = Blueprint("goal", __name__,url_prefix="/goals")
+
+# Helper functions to validate if goal_id is integer
+def valid_int(number, parameter_type):
+    try:
+        return int(number)
+    except:
+        abort(make_response({"error": f"{parameter_type} must be an int"}, 400))
+
+def get_goal_from_id(goal_id):
+    goal_id = valid_int(goal_id, "goal_id")
+    return Goal.query.get_or_404(goal_id, description="{goal not found}")
+
+#routes that use GET method
+@goal_bp.route("", methods=["GET"])
+def get_goals():
+
+    sort_query = request.args.get('sort')
+    #sorting sorterer
+
+    if sort_query == 'asc':
+            goals = Goal.query.order_by(Goal.title.asc())
+    elif sort_query == 'desc':
+            goals= Goal.query.order_by(Goal.title.desc())
```

Comment:
Style: Funky indentation:

```suggestion
    if sort_query == 'asc':
        goals = Goal.query.order_by(Goal.title.asc())
    elif sort_query == 'desc':
        goals= Goal.query.order_by(Goal.title.desc())
```